### PR TITLE
Only register templates if enabled for user

### LINKED ISSUE #2397: "Restrict usage to specific groups" does not work properly (Buggy choices remain when clicking on [+])
**Describe the bug**
When you activate the "Restrict usage to specific groups" setting to allow only certain user groups to use Collabora Online / Nextcloud Office, the user group(s) concerned cannot effectively use the office suite anymore. However, in Nextcloud Files, when you click on the [+] button, you still have the Collabora Online / Nextcloud Office document creation options. By clicking on it, we can see that : 
- it is not possible to create a file (Collabora does not open)
- that the choices no longer have a blue, green, orange icon...
So I think it's a bug; assuming I understood the purpose of the "Restrict usage to specific groups" setting, these choices should no longer be available when clicking on the [+] button.

**To Reproduce**
1. Enable "Restrict usage to specific groups" (and select a user group)
2. Log as a user belonging to the declared group
3. Notice the described problem

**Expected behavior**
"New document", "New spreadsheet" and "New presentation" shouldn't appear.

**Screenshots**
Settings : 

What you get after in the [+] button : 

What you get after trying to create a Collabora document : 

What you only should have : 

**Client details:**
 - OS: Ubuntu 20.04 LTS
 - Browser : Firefox 103
 - Device: Laptop

## Server details

**Operating system**: Ubuntu 20.04 LTS

**Web server:**Apache

**Database:** MySQL

**PHP version:** 8.0.22

**Nextcloud version:** 20.0.4

**Version of the richdocuments app** 
COOLWSD / Collabora Office 22.05.5.2
Nextcloud Office 6.2.0

**Version of Collabora Online**
Collabora Online - Built-in CODE Server 22.5.502

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
Can be provided if necessary
```

#### Browser log
```
Can be provided if necessary
```

</details>

### PR DESCRIPTION
**Pull request by juliushaertl**
Fixes #2397

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -72,8 +73,8 @@ public function register(IRegistrationContext $context): void {
 	}
 
 	public function boot(IBootContext $context): void {
-		$context->injectFn(function (ITemplateManager $templateManager, IL10N $l10n, IConfig $config, CapabilitiesService $capabilitiesService) {
-			if (empty($capabilitiesService->getCapabilities())) {
+		$context->injectFn(function (ITemplateManager $templateManager, IL10N $l10n, IConfig $config, CapabilitiesService $capabilitiesService, PermissionManager $permissionManager) {
+			if (empty($capabilitiesService->getCapabilities()) || !$permissionManager->isEnabledForUser()) {
```
Comment (Raudius) [inline lib/AppInfo/Application.php:77]: Are capabilities not already empty when the app is disabled for the user? https://github.com/nextcloud/richdocuments/pull/2328

Comment (juliushaertl) [reply lib/AppInfo/Application.php:77]: This is where Collabora capabilities and Nextcloud capabilities might cause some confusion.  The CapabilitiesService returns the ones from Collabora and does not have such a check. 

The Nextcloud capabilities (which are usually accessed by an API call in scope of the current user) are limited however.

**Review by Raudius — APPROVED** [review]

**Comment by juliushaertl** [comment]
@Raudius Feel free to merge right away if CI is happy :)

**Pull request merged**